braindecode/augmentation/base.py

```diff
@@ -0,0 +1,204 @@
+    probability : float, optional
+        Float between 0 and 1 defining the uniform probability of applying the
+        operation. Set to 1.0 by default (e.g always apply the operation).
+    random_state: int, optional
+        Seed to be used to instatiate numpy random number generator instance.
+        Used to decide whether or not to transform given the probability
+        argument. Defaults to None.
+    """
+
+    def __init__(self, operation, probability=1.0, random_state=None):
+        super().__init__()
+        if self.forward.__func__ is Transform.forward:
+            assert callable(operation), "operation should be a ``callable``."
+
+        self.operation = operation
+        assert isinstance(probability, Real), (
+            f"probability should be a ``real``. Got {type(probability)}.")
+        assert probability <= 1. and probability >= 0., \
+            "probability should be between 0 and 1."
+        self._probability = probability
+        self.rng = check_random_state(random_state)
+
+    def get_params(self, X, y):
+        return tuple()
+
+    def forward(self, X: Tensor, y: Tensor = None) -> Output:
+        """General forward pass for an augmentation transform
+
+        Parameters
+        ----------
+        X : torch.Tensor
+            EEG input example or batch.
+        y : torch.Tensor | None
+            EEG labels for the example or batch. Defaults to None.
+
+        Returns
+        -------
+        Tensor
+            Transformed inputs.
+        Tensor, optional
+            Transformed labels. Only returned when y is not equal to
+            None or omitted.
+        """
+        out_X = X.clone()
+        if y is not None:
+            out_y = y.clone()
+        else:
+            out_y = torch.zeros(X.shape[0])
+
+        # Samples a mask setting for each example whether they should stay
+        # inchanged or not
+        mask = self._get_mask(X.shape[0])
+        num_valid = mask.sum().long()
+
+        if num_valid > 0:
+            # Uses the mask to define the output
+            out_X[mask, ...], tr_y = self.operation(
+                out_X[mask, ...], out_y[mask], *self.get_params(X, y)
+            )
+            # Apply the operation defining the Transform to the whole batch
+            if type(tr_y) is tuple:
+                out_y = tuple(tmp_y[mask] for tmp_y in tr_y)
+            else:
+                out_y[mask] = tr_y
+
+        if y is not None:
+            return out_X, out_y
+        else:
+            return out_X
+
+    def _get_mask(self, batch_size=None) -> torch.Tensor:
+        """Samples whether to apply operation or not over the whole batch
+        """
+        return torch.as_tensor(
+            self.probability > self.rng.uniform(size=batch_size)
+        )
+
+    @property
+    def probability(self):
+        return self._probability
+
+    def to_dict(self):
+        """Returns a dictionary describing the transform """
```

Comment:
Maybe describe a bit more here, what is inside the dictionary and what is this function good for like why does it exist, what is it used for?

Reply:
Good question ! It's legacy code that I will remove :)

Reply:
ok great waiting for you to remove before merging then